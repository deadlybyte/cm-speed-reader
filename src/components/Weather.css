.CM-weather-forecast {
  text-transform: capitalize;
}
