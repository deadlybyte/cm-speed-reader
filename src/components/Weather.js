import React from 'react';
import './Weather.css';

const Weather = (props) => {
  const { forecast, temperature } = props;

  return (
    <span className="CM-weather-forecast">
      Weather - {forecast ? forecast : 'Unknown'}
      <span dangerouslySetInnerHTML={{ __html: temperature !== undefined ? `, ${temperature}&deg;C` : '' }}></span>
    </span>
  );
};

export default Weather;
