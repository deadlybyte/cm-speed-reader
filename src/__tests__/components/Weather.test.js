import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Weather from '../../components/Weather';

describe('Weather', () => {
  describe('render', () => {
    it('should render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Weather />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('should render Unknown forecast', () => {
      const componentUnderTest = renderer.create(<Weather />)
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });

    it('should render the forecast and temperature', () => {
      const componentUnderTest = renderer.create(<Weather forecast="Gusty, Drizzle" temperature={10} />)
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });

    it('should render the forecast', () => {
      const componentUnderTest = renderer.create(<Weather forecast="Gusty, Drizzle" />)
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });
  });
});
