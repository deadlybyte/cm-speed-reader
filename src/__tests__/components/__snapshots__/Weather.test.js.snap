// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather render should render Unknown forecast 1`] = `
<span
  className="CM-weather-forecast"
>
  Weather - 
  Unknown
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</span>
`;

exports[`Weather render should render the forecast 1`] = `
<span
  className="CM-weather-forecast"
>
  Weather - 
  Gusty, Drizzle
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</span>
`;

exports[`Weather render should render the forecast and temperature 1`] = `
<span
  className="CM-weather-forecast"
>
  Weather - 
  Gusty, Drizzle
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": ", 10&deg;C",
      }
    }
  />
</span>
`;
